docs/tutorials/python/mnist.md

```diff
@@ -1,116 +1,135 @@
+
+Image batches are commonly represented by a 4-D matrix with shape `(batch_size, num_channels, width, height)`. For MNIST dataset, since the images are grayscale, there is only one color channel. Also, the images are 28x28 pixels, and so each image has width and height equal to 28. Therefore, the shape of input is `(batch_size, 1, 28, 28)`. Another important consideration is the order of input samples. When feeding training examples, it is critical that we don't feed samples with the same label in succession. Doing so can slow down training.
+Data iterators take care of this by randomly shuffling the inputs. Note that we only need to shuffle train data. The order does not matter for test data.
+
+The following source code initializes the data iterators for MNIST dataset. Note that we initialize two iterators: one for train data and one for test data.
 
 ```python
 batch_size = 100
 train_iter = mx.io.NDArrayIter(mnist['train_data'], mnist['train_label'], batch_size, shuffle=True)
 val_iter = mx.io.NDArrayIter(mnist['test_data'], mnist['test_label'], batch_size)
 ```
 
+## Training
+We will cover a couple of approaches for performing the hand written digit recognition task. The first approach makes use of a more traditional deep neural network architecture called Multilayer Percepton. We'll discuss its drawbacks and use that as a motivation to introduce a second more advanced approach called Convolution Neural Network that has proven to work very well for image classification tasks.
+
 ## Multilayer Perceptron
 
-We first use [multilayer perceptron](https://en.wikipedia.org/wiki/Multilayer_perceptron) to solve this problem. We
-define a multilayer perceptron by using MXNet's symbolic interface. The
-following command create a place holder variable for the input data.
+The first approach makes use of a [Multilayer Perceptron](https://en.wikipedia.org/wiki/Multilayer_perceptron), MLP for short, to solve this problem. We'll define the MLP using MXNet's symbolic interface. We begin by creating a place holder variable for the input data. When working with an MLP, we need to flatten our 28x28 images into a flat 1-D structure of 784 (28 * 28) raw pixel values. The order of pixel values in the flattened vector does not matter as long as we are being consistent about how we do this across all images.
```

Comment:
Redundant since we just said that above. Maybe remove the redundant part under training. 

Just say "Multilayer Perceptron (MLP)". this is the standard way to introduce an acronym. Other wise don't capitalize the term.